engine/crates/engine-v2/src/engine.rs

```diff
@@ -70,6 +72,20 @@ impl<R: Runtime> Engine<R> {
             runtime.kv().clone(),
         );
 
+        let retry_budgets = schema
+            .iter_graphql_endpoints()
+            .map(|endpoint| {
+                let retry_config = endpoint.retry_config()?;
+
+                // Defaults: https://docs.rs/tower/0.4.13/src/tower/retry/budget.rs.html#137-139
+                let ttl = retry_config.ttl.unwrap_or(std::time::Duration::from_secs(10));
+                let min_per_second = retry_config.min_per_second.unwrap_or(10);
+                let retry_percent = retry_config.retry_percent.unwrap_or(0.2);
```

Comment:
Since the config is `None` by default we don't retry at all, but if the retry section is present, we'll retry. I think we could add an `enabled` option for clarity.